website/src/models/group.js

```diff
@@ -444,19 +444,45 @@ GroupSchema.methods.leave = function(user, keep, mainCb){
         var leader = group.leader;
 
         if(leader == user._id || !~group.members.indexOf(leader)){
-          var seniorMember = _.find(group.members, function (m) {return m != user._id});
 
-          // could not exist in case of public guild with 1 member who is leaving
-          if(seniorMember){
-            if (leader == user._id || !~group.members.indexOf(leader)) {
-              update['$set'] = update['$set'] || {};
-              update['$set'].leader = seniorMember;
+          var seniorMember = undefined;
+          var today = new Date();
+          var lastWeek = new Date(today.getFullYear(), today.getMonth(), today.getDate() - 7);
+
+          User.findOne({
+            '_id': { $in : group.members, $ne : leader },
+            'auth.blocked' : { $exists: false },
+            'flags.chatRevoked': { $exists: false },
+            'lastCron': { $gt: lastWeek }
+          })
+          .sort({'lastCron': -1})
```

Comment:
Since we're using findOne, I'm not certain that this sort function is actually doing anything. Can you dig up some documentation figuring out if it is one way or the other?